packages/react-components/src/components/Datepicker/hooks.ts

```diff
@@ -0,0 +1,95 @@
+import * as React from 'react';
+import { calculateDatePickerMonth, getInitialStateFromProps } from './helpers';
+import {
+  IRangeDatePickerProps,
+  IRangeDatePickerReducerAction,
+  IRangeDatePickerState,
+  RangeDatePickerAction,
+  RangeDatePickerReducer,
+} from './types';
+
+export const useRangeDatePickerState = (
```

Comment:
Because the component has some extended internal state I decided to use useReducer hook to handle that. The main reasons: 
- initial state is calculated from props
- choosing one date we need to control other dates as well (reset the state or as in `RangeDatePickerAction.SELECT_SECOND_DAY_AS_FROM` switch the dates